core/src/com/metalrain/stocksimulator/state/components/StockImpactEventComponent.java

```diff
@@ -11,6 +15,10 @@
     private final int[] demand_impact;
     private final String[] market_names;
     private final int[] supply_impact;
+    ///new vars
+    private String toString;
+    String demandViewData="\nDemand:   ";
+    String supplyViewData="Supply:   ";
 
     public StockImpactEventComponent(String message, int iteration, String[] market_names, int[] supply_impact, int[] demand_impact) {
```

Comment:
When constructing the event's, we probably need to use the new, random, names in the market_names array